app/src/main/java/com/humberto/tasky/agenda/data/task/TaskRepositoryImpl.kt

```diff
@@ -72,15 +104,41 @@ class TaskRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun syncPendingTasks() {
-        taskDao.getTasksPendingSync(
-            userId = localUserId ?: ""
-        ).map { taskPendingSyncEntity ->
-            safeCall {
-                agendaApi.createTask(taskPendingSyncEntity.task.toTask().toTaskRequest())
-            }.onSuccess {
-                taskDao.deleteTaskPendingSync(taskPendingSyncEntity.taskId)
-            }
+    private inline fun <reified T: ListenableWorker> enqueuePendingWorker(taskId: String) {
+        val request = OneTimeWorkRequestBuilder<T>()
+            .setInputData(
+                workDataOf("TASK_ID" to taskId)
+            )
+            .setInitialDelay(15, TimeUnit.MINUTES)
```

Comment:
Why do you want an initial delay here? These workers can be executed immediately if possible. You can add an exponential backoff policy to perform retries at increasing intervals.